parent/pom.xml

```diff
@@ -1513,7 +1505,19 @@
                <buildOutputDirectory>${basedir}/eclipse-output</buildOutputDirectory>
             </configuration>
          </plugin>
-
+         <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-checkstyle-plugin</artifactId>
+            <executions>
+               <execution>
+                  <id>checkstyle</id>
+                  <phase>validate</phase>
```

Comment:
hmmm yeah it also makes sense... when I think about it again... your approach might be slightly more friendly for those who do not have properly configured IDE... Let's go for it...